### ISSUE TITLE
app crashes with `Fatal Python error: ffi.from_handle() detected that the address passed points to garbage` shortly after starting only on Windows since 20.4.0

### ISSUE THREAD

**Issue by PawelTroka** [gevent/gevent#1587]
* gevent version: 20.4.0
* Python version: cPython 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)] on win32 downloaded from python.org
* Operating System: Microsoft Windows 10 Version 1903 (OS Build 18362.720)

### Description:

Nothing fancy just running an app with Flask, gevent and gevent-websocket (all in their latest versions) and it crashes shortly after running. Sometimes it is nearly instantly, sometimes it takes even several seconds (no longer than a minute though) but it always does exit with error code. Strangely it only happens on Windows - on Linux Ubuntu (through docker) everything is all right. Everything worked OK back on gevent 1.5.0 but this warning (https://github.com/gevent/gevent/issues/1564) was written repeatedly.

```
Fatal Python error: ffi.from_handle() detected that the address passed points to garbage. If it is really the result of ffi.new_handle(), then the Python object has already been garbage collected
Python runtime state: initialized

Thread 0x00004958 (most recent call first):
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_threading.py", line 60 in wait
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_threading.py", line 146 in get
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\threadpool.py", line 155 in run

Thread 0x00004ad8 (most recent call first):
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_threading.py", line 60 in wait
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_threading.py", line 146 in get
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\threadpool.py", line 155 in run

Current thread 0x0000549c (most recent call first):
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_ffi\loop.py", line 264 in python_stop
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\libuv\loop.py", line 498 in __run_queued_callbacks
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\libuv\loop.py", line 218 in _run_callbacks
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\_ffi\loop.py", line 300 in python_prepare_callback
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\libuv\loop.py", line 542 in run
  File "C:\Users\patr\AppData\Local\Programs\Python\Python38\lib\site-packages\gevent\hub.py", line 618 in run

Process finished with exit code -1073740791 (0xC0000409)
```

**Comment by jamadden** [comment]
Can you provide a [short, self contained example](http://sscce.org) that reliably crashes? Without being able to reproduce the problem it's very difficult to find and fix. It looks as if a watcher object is still active when it shouldn't be, but beyond that theres not much I can say.

> Strangely it only happens on Windows - on Linux Ubuntu (through docker) everything is all right. 

The default event loop is different on Windows and Linux. It would be interesting to know if [using the `libuv-cffi` loop](http://www.gevent.org/configuration.html#gevent._config.Config.loop) on Linux reproduces the crash.

**Comment by PawelTroka** [comment]
@jamadden Ok, I can try to provide a short, self contained example but this is in proprietary application, so it can take me some time to properly extract that.

**Comment by jtc42** [comment]
I'm facing the exact same issue. Gevent 20.04, Windows 10. Works fine on WSL2

**Comment by susoooon** [comment]
from gevent import monkey,pywsgi
monkey.patch_all(thread=False)
from flask import Flask,request
from flask import request, jsonify, json
app = Flask(__name__)

app.config['JSON_AS_ASCII'] = False


@app.route('/helloworld', methods=[ 'POST','GET'])
def hello():
    return "hello world"
if __name__ == '__main__':
    app.run(host="127.0.0.1",port=5000)

_________________________________________________
when i start flask and request,first time is ok,but second the error happend

**Comment by int3l** [comment]
Can confirm the same error on Windows 10 (1909) Latest with python 3.8.1 and gevent 20.5.0
As a workaround I should revert back to gevent 1.5.0 until this get sorted.

Error output for reference:
```
Fatal Python error: ffi.from_handle() detected that the address passed points to garbage. If it is really the result of ffi.new_handle(), then the Python object has already been garbage collected
Python runtime state: initialized

Thread 0x00000858 (most recent call first):
  File "Z:\Project\env\lib\site-packages\gevent\_threading.py", line 60 in wait
  File "Z:\Project\env\lib\site-packages\gevent\_threading.py", line 146 in get
  File "Z:\Project\env\lib\site-packages\gevent\threadpool.py", line 155 in run

Current thread 0x00002230 (most recent call first):
  File "Z:\Project\env\lib\site-packages\gevent\_ffi\loop.py", line 264 in python_stop
  File "Z:\Project\env\lib\site-packages\gevent\libuv\loop.py", line 498 in __run_queued_callbacks
  File "Z:\Project\env\lib\site-packages\gevent\libuv\loop.py", line 218 in _run_callbacks
  File "Z:\Project\env\lib\site-packages\gevent\_ffi\loop.py", line 300 in python_prepare_callback
  File "Z:\Project\env\lib\site-packages\gevent\libuv\loop.py", line 542 in run
  File "Z:\Project\env\lib\site-packages\gevent\hub.py", line 618 in run
```

**Comment by jamadden** [comment]
Fixed in #1602, as best I can tell.

**Comment by RafAustralia** [comment]
same issue here - a crash... but with a ffi fail error,

```
2025-06-19 12:02:55.080 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_failed/dns_server
Fatal Python error: b_from_handle: ffi.from_handle() detected that the address passed points to garbage. If it is really the result of ffi.new_handle(), then the Python object has already been garbage collected
Python runtime state: initialized
Current thread 0x00007f11a5e27b38 (most recent call first):
  File "/usr/local/lib/python3.13/site-packages/pycares/__init__.py", line 120 in _query_cb
  File "/usr/local/lib/python3.13/site-packages/pycares/__init__.py", line 424 in <lambda>
  File ???, line 0 in ???
Thread 0x00007f119bbffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119c5ffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119cbf7b38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119ccfbb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119cdffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119d3f7b38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119d4fbb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119d5ffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119db6fb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119dc73b38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119de77b38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119df7bb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119e07fb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f119ffffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a087fb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a19fbb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a1affb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a23ffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a2dffb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a33ebb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a34efb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a3efbb38 (most recent call first):
  File "/usr/local/lib/python3.13/socket.py", line 719 in readinto
  File "/usr/local/lib/python3.13/http/client.py", line 539 in _read_next_chunk_size
  File "/usr/local/lib/python3.13/http/client.py", line 579 in _get_chunk_left
  File "/usr/local/lib/python3.13/http/client.py", line 597 in _read_chunked
  File "/usr/local/lib/python3.13/http/client.py", line 473 in read
  File "/usr/local/lib/python3.13/site-packages/urllib3/response.py", line 862 in _fp_read
  File "/usr/local/lib/python3.13/site-packages/urllib3/response.py", line 879 in _raw_read
  File "/usr/local/lib/python3.13/site-packages/urllib3/response.py", line 955 in read
  File "/usr/local/lib/python3.13/site-packages/docker/api/client.py", line 365 in _stream_helper
  File "/usr/local/lib/python3.13/site-packages/docker/utils/json_stream.py", line 16 in stream_as_text
  File "/usr/local/lib/python3.13/site-packages/docker/utils/json_stream.py", line 60 in split_buffer
  File "/usr/local/lib/python3.13/site-packages/docker/api/client.py", line 360 in _stream_helper
  File "/usr/local/lib/python3.13/site-packages/docker/types/daemon.py", line 29 in __next__
  File "/usr/src/supervisor/supervisor/docker/monitor.py", line 62 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a46ffb38 (most recent call first):
  File "/usr/local/lib/python3.13/site-packages/pyudev/_util.py", line 152 in eintr_retry_call
  File "/usr/local/lib/python3.13/site-packages/pyudev/_os/poll.py", line 91 in poll
  File "/usr/local/lib/python3.13/site-packages/pyudev/_util.py", line 152 in eintr_retry_call
  File "/usr/local/lib/python3.13/site-packages/pyudev/monitor.py", line 533 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a7a34b38 (most recent call first):
  <no Python frame>
Thread 0x00007f11a5f7bb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a607fb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a7bbcb38 (most recent call first):
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 90 in _worker
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11a7cc0b38 (most recent call first):
  File "/usr/local/lib/python3.13/logging/handlers.py", line 1539 in dequeue
  File "/usr/local/lib/python3.13/logging/handlers.py", line 1590 in _monitor
  File "/usr/local/lib/python3.13/threading.py", line 992 in run
  File "/usr/local/lib/python3.13/threading.py", line 1041 in _bootstrap_inner
  File "/usr/local/lib/python3.13/threading.py", line 1012 in _bootstrap
Thread 0x00007f11ad801b28 (most recent call first):
  File "/usr/local/lib/python3.13/asyncio/selector_events.py", line 1067 in write
  File "/usr/local/lib/python3.13/site-packages/aiohttp/_websocket/writer.py", line 139 in send_frame
  File "/usr/local/lib/python3.13/site-packages/aiohttp/client_ws.py", line 245 in send_str
  File "/usr/local/lib/python3.13/site-packages/aiohttp/client_ws.py", line 261 in send_json
  File "/usr/src/supervisor/supervisor/homeassistant/websocket.py", line 91 in async_send_command
  File "/usr/src/supervisor/supervisor/homeassistant/websocket.py", line 269 in async_send_message
  File "/usr/src/supervisor/supervisor/homeassistant/websocket.py", line 301 in async_supervisor_event_custom
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89 in _run
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2034 in _run_once
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 677 in run_forever
  File "/usr/src/supervisor/supervisor/__main__.py", line 81 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main
Extension modules: isal.igzip_lib, isal.isal_zlib, multidict._multidict, yarl._quoting_c, propcache._helpers_c, _brotli, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket.mask, aiohttp._websocket.reader_c, _cffi_backend, yaml._yaml, ciso8601, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, dbus_fast.signature, dbus_fast._private.marshaller, dbus_fast.message, dbus_fast.service, dbus_fast._private.address, dbus_fast.unpack, dbus_fast.message_bus, dbus_fast._private.unmarshaller, dbus_fast.aio.message_reader, markupsafe._speedups (total: 26)
[02:02:55] INFO: Supervisor restart after closing
2025-06-19 02:02:56.625 INFO (MainThread) [__main__] Initializing Supervisor setup
```
